### PR TITLE
14.0 add shopinvader customer delivery

### PR DESCRIPTION
**Pull request by mourad-ehm**

### CODE REVIEW

**Comment by mourad-ehm** [comment]
@hparfr

File: shopinvader_customer_delivery/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Shopinvader Invoice",
+    "summary": """Shopinvader Invoice module""",
+    "author": "Akretion",
+    "website": "https://github.com/shopinvader/odoo-shopinvader",
+    "category": "shopinvader",
+    "version": "14.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["account", "shopinvader"],
```
Comment (hparfr) [inline shopinvader_customer_delivery/__manifest__.py]: account?

File: shopinvader_customer_delivery/services/stock_move.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+from odoo.osv import expression
+
+from odoo.addons.component.core import Component
+
+
+class DeliveryMoveService(Component):
+    """Service for getting information on stock.picking"""
+
+    _inherit = ["base.shopinvader.service", "abstract.shopinvader.download"]
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py]: why download ?

File: shopinvader_customer_delivery/services/stock_move.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        domain (to have 0 picking as result)
+        :return:
+        """
+        if not self._is_logged_in():
+            return expression.FALSE_DOMAIN
+        sale_service = self.component(usage="sales")
+        sale_domain = sale_service._get_base_search_domain()
+        print(sale_domain)
+        sales = self.env[sale_service._expose_model].search(sale_domain)
+        print([("sale_id", "in", sales.ids)] + self._get_allowed_stock_move_domain())
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py]: print ?!

**Review by hparfr — CHANGES_REQUESTED** [review]

File: shopinvader_customer_delivery/__manifest__.py (hunk 1 above)
Comment (hparfr) [reply shopinvader_customer_delivery/__manifest__.py]: still this one

File: shopinvader_customer_delivery/services/stock_move.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        """Domain to filter stock move.
+
+        :return: list of tuples
+        """
+        states = ["waiting", "confirmed", "partially_available", "assigned"]
+        # todo add method to backend to get visible states
+        domain = [
+            ("picking_id.picking_type_id.code", "=", "outgoing"),
+        ]
+        if states:
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py]: useless condition

File: shopinvader_customer_delivery/services/stock_move.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        """
+        to_parse = [
+            "id:product_id",
+            "product_tmpl_id:template_id",
+            "name",
+            "default_code",
+        ]
+        return to_parse
+
+    def _add_product_info(self, stok_move):
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py]: typo

File: shopinvader_customer_delivery/services/stock_move.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+        """
+        Add info about the related product (using product_id field).
+        :param picking: stock.move
+        :return: dict
+        """
+        sale_order = stok_move.product_id
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py]: wrong  variable name

File: shopinvader_customer_delivery/services/stock_move.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        if not sale_order:
+            return {}
+        parser = self._get_parser_sale_order()
+        values = sale_order.jsonify(parser, one=True)
+        return values
+
+    def _get_parser_sale_order(self):
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py:203]: wrong variable name

File: shopinvader_customer_delivery/services/stock_move.py (hunk 6 above)
Comment (hparfr) [reply shopinvader_customer_delivery/services/stock_move.py]: @mourad-ehm please fix !

**Review by hparfr — CHANGES_REQUESTED** [review]

File: shopinvader_customer_delivery/services/stock_move.py [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+        """
+        to_parse = [
+            "id:product_id",
+            "product_tmpl_id:template_id",
+            "name",
+            "default_code",
+        ]
+        return to_parse
+
+    def _add_product_info(self, stok_move):
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py]:
```suggestion
    def _add_product_info(self, stock_move):
```

File: shopinvader_customer_delivery/services/stock_move.py [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+        Add info about the related product (using product_id field).
+        :param picking: stock.move
+        :return: dict
+        """
+        sale_order = stok_move.product_id
+        if not sale_order:
+            return {}
+        parser = self._get_parser_sale_order()
+        values = sale_order.jsonify(parser, one=True)
+        return values
```
Comment (hparfr) [inline shopinvader_customer_delivery/services/stock_move.py]:
```suggestion
        if not product:
            return {}
        parser = self._get_parser_product()
        values = product.jsonify(parser, one=True)
        return values
```

File: shopinvader_customer_delivery/services/stock_move.py [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+        ]
+        return to_parse
+
+    def _add_product_info(self, stok_move):
+        """
+        Add info about the related product (using product_id field).
+        :param picking: stock.move
+        :return: dict
+        """
+        sale_order = stok_move.product_id
```
Comment (mourad-ehm) [inline shopinvader_customer_delivery/services/stock_move.py]:
```suggestion
        sale_order = stock_move.product_id
```

File: shopinvader_customer_delivery/services/stock_move.py (hunk 7 above)
Comment (mourad-ehm) [reply shopinvader_customer_delivery/services/stock_move.py:203]: Which ?

File: shopinvader_customer_delivery/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Shopinvader Delivery",
+    "summary": """Shopinvader Invoice module""",
+    "author": "Akretion",
+    "website": "https://github.com/shopinvader/odoo-shopinvader",
+    "category": "shopinvader",
+    "version": "14.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["account", "shopinvader"],
```
Comment (mourad-ehm) [inline shopinvader_customer_delivery/__manifest__.py]:
```suggestion
    "depends": ["stock", "shopinvader"],
```

Comment (mourad-ehm) [inline shopinvader_customer_delivery/__manifest__.py]:
```suggestion
    "depends": ["stock", "shopinvader"],
```

File: shopinvader_customer_delivery/services/stock_move.py [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+
+        :return: list of tuples
+        """
+        states = ["waiting", "confirmed", "partially_available", "assigned"]
+        # todo add method to backend to get visible states
+        domain = [
+            ("picking_id.picking_type_id.code", "=", "outgoing"),
+        ]
+        if states:
+            domain.append(("state", "in", states))
```
Comment (mourad-ehm) [inline shopinvader_customer_delivery/services/stock_move.py]:
```suggestion
```

File: shopinvader_customer_delivery/services/stock_move.py [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+        """Domain to filter stock move.
+
+        :return: list of tuples
+        """
+        states = ["waiting", "confirmed", "partially_available", "assigned"]
+        # todo add method to backend to get visible states
+        domain = [
+            ("picking_id.picking_type_id.code", "=", "outgoing"),
+        ]
+        if states:
```
Comment (mourad-ehm) [inline shopinvader_customer_delivery/services/stock_move.py]:
```suggestion
        domain.append(("state", "in", states))
```

**Pull request closed without merging**